rules/apps.libsonnet

```
@@ -216,7 +238,7 @@
                     "replicaset", "$1", "owner_name", "(.*)"
                   ) * on(replicaset, namespace) group_left(owner_name) topk by(replicaset, namespace) (
                     1, max by (replicaset, namespace, owner_name) (
-                      kube_replicaset_owner{%(kubeStateMetricsSelector)s}
+                      kube_replicaset_owner{%(kubeStateMetricsSelector)s, owner_kind!=""}
```

You'll need to specifically target `Deployment` here to avoid catching other types of workload

```suggestion
                      kube_replicaset_owner{%(kubeStateMetricsSelector)s, owner_kind="Deployment"}
```